### ISSUE TITLE
An issue when creating the manifest after downloading exe and portable

### ISSUE THREAD

**Issue by BrandonWanHuanSheng** [microsoft/winget-create#382]
### Brief description of your issue

When I run wingetcreate new, creating a new manifest for yt-dlp.yt-dlp.nighty,. Specifying 4 URL at the same times. 
It look like wingetcreate throw this error.

### Steps to reproduce

wingetcreate new "https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/download/2023.06.04.142854/yt-dlp.exe" "https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/download/2023.06.04.142854/yt-dlp_x86.exe" "https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/download/2023.06.04.142854/yt-dlp_win.zip" "https://github.com/yt-dlp/yt-dlp-nightly-builds/releases/download/2023.06.04.142854/yt-dlp_win.zip"
Typing the app information.

### Expected behavior

Expected to be manifest validation succeed.

### Actual behavior

It throw this error out while creating the manifest and after downloading the 4 URL.

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentException: Object of type 'System.Linq.Enumerable+SelectListIterator`2[System.String,System.String]' cannot be converted to type 'System.Collections.Generic.List`1[System.String]'.
   at System.RuntimeType.TryChangeType(Object value, Binder binder, CultureInfo culture, Boolean needsSpecialCast)
   at System.Reflection.MethodBase.CheckArguments(StackAllocedArguments& stackArgs, ReadOnlySpan`1 parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
   at System.Reflection.RuntimeMethodInfo.InvokeOneParameter(Object obj, BindingFlags invokeAttr, Binder binder, Object parameter, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
   at Microsoft.WingetCreateCLI.PromptHelper.PromptList[T](String message, Object model, String memberName, IEnumerable`1 instance, Int32 minimum, Object validationModel, String validationName)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Microsoft.WingetCreateCLI.PromptHelper.PromptPropertyAndSetValue[T](Object model, String memberName, T instance, Int32 minimum, Object validationModel, String validationName)
   at Microsoft.WingetCreateCLI.Commands.NewCommand.PromptOptionalProperties[T](T manifest)
   at Microsoft.WingetCreateCLI.Commands.NewCommand.PromptPropertiesAndDisplayManifests(Manifests manifests)
   at Microsoft.WingetCreateCLI.Commands.NewCommand.Execute()
   at Microsoft.WingetCreateCLI.Program.Main(String[] args)

Log: 
[WingetCreateLog-2023-06-05-22-43.199.txt](https://github.com/microsoft/winget-create/files/11654035/WingetCreateLog-2023-06-05-22-43.199.txt)

### Environment

```shell
wingetcreate
Windows Package Manager Manifest Creator v1.2.6.0

winget --info
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.3030
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by BrandonWanHuanSheng** [comment]
[WingetCreateLog-2023-06-06-17-30.097.txt](https://github.com/microsoft/winget-create/files/11663037/WingetCreateLog-2023-06-06-17-30.097.txt)

**Comment by ryfu-msft** [comment]
@BrandonWanHuanSheng, this issue has been resolved by #390 and the fix is now available in our latest release[ v1.2.7.0](https://github.com/microsoft/winget-create/releases/tag/v1.2.7.0).
